### ISSUE TITLE
check that all must-have features are implemented

### ISSUE THREAD

**Issue by tessav** [CS2103AUG2016-F10-C3/main#59]
Must-have features:<br>

Support for events (i.e., has a start time and end time), deadlines (tasks that have to be done before a specific deadline), and floating tasks (tasks without specific times). Note that these category names are temporary names only. You should figure out what are the best names for these categories. Are these names intuitive to end-users? Should they even care about consciously categorizing tasks?<br>
CRUD (i.e., Create, Read, Update, Delete) support for tasks.<br>
Undo operations (at least for the most recent action)<br>
Some flexibility in the command format: Support a few natural variations of the command format.<br>
Simple search: A simple text search for finding an item if the user remembers some keywords from the item description.<br>
Some way to keep track of which items are done and which are yet to be done.<br>
The ability to specify a specific folder as the data storage location. With this feature, Jim can choose to store the data file in a local folder controlled by a cloud syncing service (e.g. dropbox), allowing him to access task data from multiple computers.<br>

**Comment by tessav** [comment]
Nice to have features : (implement 0 or more of these features)<br>

**Difficulty-low**<br>

AutomatedTesting : Achieve a very high level of automated testing.<br>
RecurringTasks: Support for managing recurring tasks. e.g. CS2103 lectures are a recurring task from week 1 to week 13 except in recess week.<br>

**Difficulty-medium**<br>

PowerSearch : More powerful and intelligent search. e.g. search for empty slots, near-match search, auto-complete (similar to Google search box), filters for other attributes (e.g. start time).<br>
FlexiCommands: High flexibility in command format. e.g., non-strict ordering of keywords, ability to specify aliases for commands, support for more ‘natural’ language input, multiple undo/redo, How flexible should the format be? As flexible, intuitive, and user-friendly as you can make it.<br>

**Difficulty-high**<br>

GoodGui: A good GUI to give visual feedback to the user e.g., in-built guidance for new users, good visual feedback for actions, feedback while typing commands, hotkey to activate/hide, notifications, etc. conveying more information visually (e.g. more important/urgent tasks stand out from the rest, longer events look different from shorter events, easy to see free slots, etc.)<br>
GoogleIntegration: Google Calendar integration. e.g. upload todo items to GCal, two-way sync, Support for GCal quick add command format, etc. You may also use the Google Tasks API for this.
Propose your own (subject to approval)<br>
Note that each student must implement at least one feature and the average amount of code one member is expected to write for the project is 0.5-1.5 KLoC.<br>

**Comment by tessav** [comment]
must have features are roughly all done
